UNIXFSv1.md

```diff
@@ -0,0 +1,487 @@
+
+This mean we deal with protobuf inside protobuf. This is for history reasons.
+This spec sometimes show this a one object for clarity reasons, when that happen it is implied that the `PBNode.Data` field is encoded into prototbuf.
+
+### Glossary
+
+- Node, Block
+  A node is a word from graph theory, this is the smallest unit present in the graph.
+  Due to how unixfs work, there is a 1 to 1 mapping between nodes and blocks.
+- File
+  A file is some container over an arbitrary sized amounts of bytes.
+  Files can be said to be single block, or multi block, in the later case they are the concatenation of multiple children files.
+- Directory, Folder
+  This is a named collection of children nodes.
+- HAMT Directory
+  This is a Hashed-Array-Mapped-Tree datastructure representing a Directory, thoses are used to split directories into multiple blocks when they get too big.
+- Symlink
+  This represent a POSIX Symlink.<!--TODO: Add link to POSIX spec.-->
+
+### Paths
+
+Path is a collection of path component (some bytes) seperated by `/` 0x2f, red from left to right, it is inspired by POSIX paths.
+
+Components MUST NOT contain `/` characters because else it would break the path into two components.
+
+Components SHOULD be UTF8 unicode.
+
+### Escaping
+
+The `\` may be supposed to trigger an escape sequence.
+
+This might be a thing, but is broken and inconsistent current implementations.
+So until we agree on a new spec for this, you SHOULD NOT use any escape sequence and non ascii character.
+
+
+### SHOULD NOT names
+
+Thoses names SHOULD NOT<!--MUST NOT ? in future revisions--> be used:
+
+- `.` (as this represent the self node in POSIX pathing)
+- `..` (as this represent the parrent node in POSIX pathing)
+- nothing (<!--TODO: check that this is true-->we don't actually know the failure mode for this, but it really feels like this shouldn't be a thing)
+
+## How to read a Node
+
+First you get some CID, this will be what will we be trying to decode.
+
+For recap, every CIDs MUST include:
+1. A [multicodec](https://github.com/multiformats/multicodec), also called codec.
+1. A [Multihash](https://github.com/multiformats/multihash) (used to specify a hashing algorithm, some hashing parameters and some digest)
+
+### Get the block
+
+The first step is to get the block, that means the actual bytes which hashed by the multihash inside the CID give you the same multihash back.
+
+### Decoding the bytes
+
+#### Block limit
+
+A so called "block limit" is in place, we do not allow any single block to be bigger than 2MiB.
+
+Implementation SHOULD try to not emit 1MiB bigger blocks, but MUST decode blocks <= 2MiB.
+
+#### Multicodecs
+
+With Unixfs we deal with two codecs which will be decoded differently:
+- `Raw`, single blocks files
```

Comment:
there are inconsistencies in the capitalisation of these codecs.
should these be `dag-pb`, `Dag-PB`, `DAG-PB`, others?

Reply:
^ Ditto on this, I noticed this at the beginning of the doc